framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -453,9 +374,97 @@ AutomaticMortarGeneration::buildMortarSegmentMesh()
     if (orientation1_valid && orientation2_valid)
       throw MooseException(
           "AutomaticMortarGeneration: Both orientations cannot simultaneously be valid.");
+
+    // We are going to treat the case where both orientations are invalid as a case in which we
+    // should not be splitting the mortar mesh to incorporate primary mesh elements.
+    // In practice, this case has appeared for very oblique projections, so we assume these cases
+    // will not be considered in mortar thermomechanical contact.
     if (!orientation1_valid && !orientation2_valid)
-      throw MooseException(
-          "AutomaticMortarGeneration: Both orientations cannot simultaneously be invalid.");
+    {
+      mooseDoOnce(mooseWarning(
+          "AutomaticMortarGeneration: Both orientations cannot simultaneously be invalid. "
+          "This can indicate there are very oblique projections between primary (mortar) "
+          "and secondary (non-mortar) surfaces for a good problem set up. It can also mean your "
+          "time step is too large. This message is only printed once."));
```

Comment:
```suggestion
          "AutomaticMortarGeneration: Unable to determine valid secondary-primary orientation. "
          "Consequently we will consider projection of the primary node invalid and not split the mortar segment. "
          "This situation can indicate there are very oblique projections between primary (mortar) "
          "and secondary (non-mortar) surfaces for a good problem set up. It can also mean your "
          "time step is too large. This message is only printed once."));
```